### PR TITLE
sriov case auto 

### PR DESCRIPTION
**Pull request by zhaozhanqi**
Draft one pr to auto sriov case
seems jenkins is down, run the in my local
http://pastebin.test.redhat.com/876540
@anuragthehatter @huiran0826 @rbbratta @weliang1 PTAL, thanks

### CODE REVIEW

File: features/networking/sriov.feature [hunk 1]
```diff
@@ -14,3 +14,58 @@ Feature: Sriov related scenarios
...
+  # @case_id OCP-24702
+  @destructive
+  @admin
+  Scenario: netdevice VF for XXV710 can be worked well when sriovnetworknodepolicies is created for rhcos node
+    Given the sriov operator is running well
+    #When I run the :create admin command with:
+    #  | f | <%= BushSlicer::HOME %>/testdata/networking/sriov/sriovnetworkpolicy/intel-netdevice.yaml |
+    #Then the step should succeed
+    Given I create resource with following:
+       | cr_yaml       |  <%= BushSlicer::HOME %>/testdata/networking/sriov/sriovnetworkpolicy/intel-netdevice.yaml |
```
Comment (pruan-rht) [inline features/networking/sriov.feature:29]: @zhaozhanqi pls use `Given I obtain...` to get the testdata going forward.

File: features/networking/sriov.feature [hunk 2]
```diff
@@ -14,3 +14,58 @@ Feature: Sriov related scenarios
...
+      | namespace   | openshift-sriov-network-operator |
+      | o           | yaml                             |
+    Then the step should succeed
+    And the output should contain "vfID: 4"
+    """
+    Given I switch to the first user
+    And I have a project
+    And evaluation of `project.name` is stored in the :usr_project clipboard
+    Given I create resource with following:
+       | cr_yaml       |  <%= BushSlicer::HOME %>/testdata/networking/sriov/sriovnetwork/static-sriovnetwork.yaml |
```
Comment (pruan-rht) [inline features/networking/sriov.feature:46]: @zhaozhanqi pls use Given I obtain... to get the testdata going forward.

File: features/networking/sriov.feature [hunk 3]
```diff
@@ -14,3 +14,58 @@ Feature: Sriov related scenarios
...
+    When I run the :get admin command with:
+      | resource  | net-attach-def        |
+      | namespace | <%= cb.usr_project%>  |
+    Then the step should succeed
+    And the output should contain "static-sriovnetwork"
+    """ 
+    Given I switch to the first user
+    And I use the "<%= cb.usr_project%>" project
+    When I run the :create client command with:
+      | f | <%= BushSlicer::HOME %>/testdata/networking/sriov/pod/sriov-without-resource-static.yaml |
```
Comment (pruan-rht) [inline features/networking/sriov.feature:63]: @zhaozhanqi pls use Given I obtain... to get the testdata going forward.

File: features/step_definitions/sriov.rb [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+Given /^sriov config daemon is ready$/ do
+  ensure_admin_tagged
+  project("openshift-sriov-network-operator")
+  step %Q/a pod becomes ready with labels:/, table(%{
+    | app=sriov-network-config-daemon |
+  })
+end
+
+Given /^I create resource with following:$/ do | table |
```
Comment (pruan-rht) [inline features/step_definitions/sriov.rb:57]: @zhaozhanqi this step looks like is `sriov` speciifc?  If that's the case, we should change the step name to indicate that. `I create resource` is too generic.

File: features/networking/sriov.feature (hunk 1 above)
Comment (zhaozhanqi) [reply features/networking/sriov.feature:29]: thanks @pruan-rht updated.

File: features/networking/sriov.feature (hunk 2 above)
Comment (zhaozhanqi) [reply features/networking/sriov.feature:46]: updated. thanks

File: features/networking/sriov.feature (hunk 3 above)
Comment (zhaozhanqi) [reply features/networking/sriov.feature]: updated. thanks

File: features/step_definitions/sriov.rb (hunk 4 above)
Comment (zhaozhanqi) [reply features/step_definitions/sriov.rb]: thanks your advice @pruan-rht , how about `I create sriov resource with following`

**Comment by zhaozhanqi** [comment]
@anuragthehatter @rbbratta @huiran0826 @weliang1  any comments for this?

File: features/step_definitions/sriov.rb [hunk 5]
```diff
@@ -0,0 +1,122 @@
+Given /^the sriov operator is running well$/ do
+  ensure_destructive_tagged
+  ensure_admin_tagged
+  step %Q/I switch to cluster admin pseudo user/
+  step %Q/evaluation of `cluster_version('version').channel.split('-')[1]` is stored in the :ocp_cluster_version clipboard/
+
+  unless project('openshift-sriov-network-operator').exists?
+    if cb.ocp_cluster_version.include? "4.[34]."
+      # create 4.3/4.4 namespaces
+      namespace_yaml ||= "#{ENV['BUSHSLICER_HOME']}/testdata/networking/sriov/namespace/43/ns.yaml"
```
Comment (pruan-rht) [inline features/step_definitions/sriov.rb:10]: @zhaozhanqi please use the `I obtain the datafile` step

File: features/step_definitions/sriov.rb [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+      @result = admin.cli_exec(:create, f: namespace_yaml)
+      raise "Error creating namespace for openshift-sriov-network-operator" unless @result[:success]
+    else
+      # Create namespace for 4.5 and later
+      namespace45_yaml ||= "#{ENV['BUSHSLICER_HOME']}/testdata/networking/sriov/namespace/45/ns45.yaml"
```
Comment (pruan-rht) [inline features/step_definitions/sriov.rb:15]: @zhaozhanqi please use the I obtain the datafile step

File: features/step_definitions/sriov.rb [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+      @result = admin.cli_exec(:create, f: namespace45_yaml)
+      raise "Error creating namespace for openshift-sriov-network-operator" unless @result[:success]
+    end
+    
+  end
+  step %Q/I use the "openshift-sriov-network-operator" project/
+  unless operator_group('sriov-network-operators').exists?
+    og_yaml ||= "#{ENV['BUSHSLICER_HOME']}/testdata/networking/sriov/og/og.yaml"
```
Comment (pruan-rht) [inline features/step_definitions/sriov.rb:23]: @zhaozhanqi please use the I obtain the datafile step

File: features/step_definitions/sriov.rb [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+    @result = admin.cli_exec(:create, f: og_yaml)
+    raise "Error creating og for openshift-sriov-network-operator" unless @result[:success]
+  end
+  
+  unless subscription('sriov-network-operator-subsription').exists?
+     sub_yaml ||= "#{ENV['BUSHSLICER_HOME']}/testdata/networking/sriov/subscription/sub.yaml"
```
Comment (pruan-rht) [inline features/step_definitions/sriov.rb:29]: @zhaozhanqi please use the I obtain the datafile step

File: features/step_definitions/sriov.rb [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+  when "sriovnetwork"
+    cr_project = opts[:project]
+    logger.info("resource type is: sriovnetwork")
+    if sriov_network(cr_name).exists?
+       step %Q/I delete the "#{cr_name}" sriovnetwork/
+    end
+
+    #@result = admin.cli_exec(:create, f: resource_yaml, n: sriov_ns)
+     step %Q/I process and create:/, table(%{
+          | f | #{resource_yaml}                   |
```
Comment (pruan-rht) [inline features/step_definitions/sriov.rb:88]: please remove extra spaces

File: features/step_definitions/sriov.rb (hunk 5 above)
Comment (zhaozhanqi) [reply features/step_definitions/sriov.rb:10]: @pruan-rht like the following steps?
```
  step %Q{I obtain test data file "networking/sriov/namespace/43/ns.yaml"}
  ns = YAML.load(@result[:response])
  namespace_yaml = @result[:abs_path]
```

Comment (pruan-rht) [reply features/step_definitions/sriov.rb:10]: @zhaozhanqi you just need to call ```step %Q{I obtain test data file "networking/sriov/namespace/43/ns.yaml"}```.  After that execution, you can then reference `ns.yaml`

File: features/step_definitions/sriov.rb (hunk 9 above)
Comment (zhaozhanqi) [reply features/step_definitions/sriov.rb]: updated. thanks

File: features/step_definitions/sriov.rb (hunk 8 above)
Comment (zhaozhanqi) [reply features/step_definitions/sriov.rb]: updated. thanks

File: features/step_definitions/sriov.rb (hunk 7 above)
Comment (zhaozhanqi) [reply features/step_definitions/sriov.rb]: @pruan-rht updated. thanks

File: features/step_definitions/sriov.rb (hunk 6 above)
Comment (zhaozhanqi) [reply features/step_definitions/sriov.rb]: @pruan-rht updated. thanks

File: features/step_definitions/sriov.rb (hunk 5 above)
Comment (zhaozhanqi) [reply features/step_definitions/sriov.rb]: @pruan-rht updated. thanks

**Pull request merged**